RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToClusterKernel.dev.cc

```diff
@@ -570,12 +572,31 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
           const auto elementsPerBlockFindClus = FindClus<TrackerTraits>::maxElementsPerBlock;
           const auto workDivMaxNumModules = cms::alpakatools::make_workdiv<Acc1D>(blocks, elementsPerBlockFindClus);
 
+          // allocate a transient collection for the fake pixels recovered by the digi morphing algorithm
+          auto fakes_d = SiPixelDigisSoACollection(blocks * digiMorphingConfig.maxFakesInModule, queue);
+
 #ifdef GPU_DEBUG
           std::cout << " FindClus kernel launch with " << numberOfModules << " blocks of " << elementsPerBlockFindClus
                     << " threadsPerBlockOrElementsPerThread\n";
 #endif
-          alpaka::exec<Acc1D>(
-              queue, workDivMaxNumModules, FindClus<TrackerTraits>{}, digis_d->view(), clusters_d->view(), wordCounter);
+
+          auto morphingModules_d =
+              cms::alpakatools::make_device_buffer<uint32_t[]>(queue, digiMorphingConfig.morphingModules.size());
+          auto morphingModules_h = cms::alpakatools::make_host_view(digiMorphingConfig.morphingModules.data(),
+                                                                    digiMorphingConfig.morphingModules.size());
+          alpaka::memcpy(queue, morphingModules_d, morphingModules_h);
```

Comment:
This is going to stay the same for the whole job.
Can you move the allocation of the buffer and the copy outside of the event loop ?
One option would be to use the `cms::alpakatools::MoveToDeviceCache` functionality (see the [README.md](https://github.com/cms-sw/cmssw/blob/master/HeterogeneousCore/AlpakaCore/README.md#copy-eg-configuration-data-to-all-devices-in-edproducer)).